jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/AbstractLabelEnricher.java

```diff
@@ -133,14 +164,26 @@ public void enrich(PlatformMode platformMode, KubernetesListBuilder builder) {
       @Override
       public void visit(ObjectMetaBuilder element) {
         Map<String, String> labels = Optional.ofNullable(element.getLabels()).orElse(new HashMap<>());
-        MapUtil.mergeIfAbsent(labels, createLabels(false));
+        MapUtil.mergeIfAbsent(labels, createLabels(false, Collections.emptyMap()));
         element.withLabels(labels);
       }
     });
   }
 
-  private Map<String, String> mergedSelectors(Map<String, String> originalSelectors) {
-    MapUtil.mergeIfAbsent(originalSelectors, createLabels(true));
+  @SuppressWarnings("unchecked")
+  private Map<String, String> mergedSelectors(Map<String, String> originalSelectors, List<Supplier<Properties>> labelSupplierList) {
```

Comment:
I'm still not sure why we're using Supplier<Properties> here instead of providing the properties themselves. Is there any reason to make this lazy?